.. _HPCBIOS_2012-94:

HPCBIOS_2012-94: Bioinformatics & Comp. Biology Productivity Environment
================================================================================

.. note::

  Bioinformatics & Computational Biology productivity environment includes a set of HPC tools
  which are needed for scientific computing and visualization in the respective domain. 
  The following is an attempt to define which ones are relevant for the HPCBIOS communities and set an action plan.

  * BC Policy: HPCBIOS_2012-94
  * Date of Policy: 1st December 2012

The main objective of this policy is to provide the following common
productivity environment across any HPCBIOS resources:

+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| Name                                   | Prefered version(s)         | Compliance level   | Reference URL                                              |
+========================================+=============================+====================+============================================================+
| **Python**                             | v2.7.3                      | MUST               | http://www.python.org/                                     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **scipy**                              | 0.10.1                      | MUST               | http://www.scipy.org                                       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **numpy**                              | 1.6.1                       | MUST               | http://numpy.scipy.org                                     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Cython**                             | 0.16                        | MUST               | http://pypi.python.org/pypi/Cython/                        |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| BioPython                              | latest                      | SHOULD             | http://biopython.org                                       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| R language                             | v2.14.2 or v2.15.0          | SHOULD             | http://www.r-project.org                                   |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| Bioconductor                           | v2.10                       | SHOULD             | http://www.bioconductor.org                                |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| Ruby                                   | v1.9.3-p194                 | SHOULD             | http://www.python.org/                                     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| BioRuby                                | latest                      | SHOULD             | http://bioruby.org                                         |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| Perl                                   | v5.x                        | SHOULD             | http://www.perl.org/                                       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| BioPerl                                | latest                      | SHOULD             | http://www.bioperl.org/                                    |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **BLAST+**                             | v2.2.26 or later            | MUST               | http://blast.ncbi.nlm.nih.gov/                             |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **HMMER**                              | v3                          | MUST               | http://hmmer.janelia.org/                                  |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **MrBayes**                            | v3.1.2                      | MUST               | http://mrbayes.sourceforge.net                             |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **AMOS**                               | 3.1.0                       | SHOULD             | http://sourceforge.net/apps/mediawiki/amos                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Boost**                              | 1.51                        | SHOULD             | http://www.boost.org                                       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Bowtie2**                            | 2.0.0-beta7                 | SHOULD             | http://bowtie-bio.sourceforge.net/bowtie2                  |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **BWA**                                | 0.6.2                       | MUST               | http://bio-bwa.sourceforge.net                             |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **ClustalW2**                          | 2.1                         | MUST               | http://www.ebi.ac.uk/Tools/msa/clustalw2                   |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Cufflinks**                          | 2.0.2                       | SHOULD             | http://cufflinks.cbcb.umd.edu                              |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Eigen**                              | 3.1.1                       | SHOULD             | http://eigen.tuxfamily.org/index.php?title=Main_Page       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **FASTX-Toolkit**                      | 0.0.13.2                    | SHOULD             | http://hannonlab.cshl.edu/fastx_toolkit                    |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **libgtextutils**                      | 0.6.1                       | SHOULD             | http://hannonlab.cshl.edu/fastx_toolkit                    |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **IDBA-UD**                            | 1.0.9                       | SHOULD             | http://i.cs.hku.hk/~alse/hkubrg/projects/idba_ud           |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Infernal**                           | 1.1rc1                      | SHOULD             | http://infernal.janelia.org                                |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **MUMmer**                             | 3.23                        | SHOULD             | http://mummer.sourceforge.net                              |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **MetaVelvet**                         | 1.2.01                      | SHOULD             | http://metavelvet.dna.bio.keio.ac.jp                       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Oases**                              | 0.2.08                      | SHOULD             | http://www.ebi.ac.uk/~zerbino/oases                        |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **pandaseq**                           | 2.1                         | SHOULD             | https://github.com/neufeld/pandaseq                        |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **RNAz**                               | 2.1                         | SHOULD             | http://www.tbi.univie.ac.at/~wash/RNAz                     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **SOAPdenovo**                         | 1.05                        | SHOULD             | http://soap.genomics.org.cn/index.html                     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **SAMtools**                           | 0.1.18                      | SHOULD             | http://samtools.sourceforge.net                            |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **TopHat**                             | 2.0.4                       | SHOULD             | http://tophat.cbcb.umd.edu                                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **ViennaRNA**                          | 2.0.7                       | SHOULD             | http://www.tbi.univie.ac.at/~ronny/RNA/vrna2.html          |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **Velvet**                             | 1.2.07                      | SHOULD             | http://www.ebi.ac.uk/~zerbino/velvet                       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **BFAST**                              | 0.7.0a                      | SHOULD             | http://bfast.sourceforge.net/                              |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **PolyPhen-2**                         | 2.2.2r405                   | SHOULD             | http://genetics.bwh.harvard.edu/pph2/dokuwiki/start        |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **PLINK**                              | 1.07                        | SHOULD             | http://pngu.mgh.harvard.edu/purcell/plink/                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **CD-HIT**                             | 4.5.5                       | SHOULD             | http://weizhong-lab.ucsd.edu/cd-hit/                       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **MCL**                                | 12.135                      | SHOULD             | http://micans.org/mcl/                                     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **MEME**                               | 4.8.0                       | SHOULD             | http://meme.nbcr.net/                                      |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| **SSAHA**                              | 31c                         | SHOULD             | http://www.sanger.ac.uk/resources/software/ssaha/          |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| NCBI toolkit                           | latest                      | SHOULD             | http://www.ncbi.nlm.nih.gov/toolkit                        |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| igblast                                | latest                      | SHOULD             | ftp://ftp.ncbi.nlm.nih.gov                                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| MPI-Blast                              | latest                      | SHOULD             | http://cbrc.musc.edu/roll-documentation/bio/4.3/           |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| mpiBLAST                               | latest                      | SHOULD             | http://www.mpiblast.org/                                   |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| GPU-BLAST                              | latest                      | SHOULD             | http://eudoxus.cheme.cmu.edu/gpublast/gpublast.html        |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| CUDA-BLASTP                            | latest                      | SHOULD             | https://sites.google.com/site/liuweiguohome/software       |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| mpiCUDA-BLASTP                         | latest                      | SHOULD             | https://sites.google.com/site/liuweiguohome/mpicuda-blastp |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| mr-mpi-blast                           | latest                      | SHOULD             | https://github.com/andreyto/mr-mpi-blast ???               |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+


----

Future potential candidates for this policy:

+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| pysam                                  | latest                      | SHOULD             | http://code.google.com/p/pysam/                            |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| MUSCLE                                 | latest                      | SHOULD             | http://www.drive5.com/muscle/                              |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| ReadSeq                                | latest                      | SHOULD             | http://iubio.bio.indiana.edu/soft/molbio/readseq/java/     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| RepeatMasker                           | latest                      | SHOULD             | http://www.repeatmasker.org/                               |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| annovar                                | latest                      | SHOULD             | http://www.openbioinformatics.org/annovar/                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| phrap                                  | latest                      | SHOULD             | http://www.phrap.org/phredphrapconsed.html                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| phred                                  | latest                      | SHOULD             | http://www.phrap.org/phredphrapconsed.html                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| consed                                 | latest                      | SHOULD             | http://www.phrap.org/consed/consed.html                    |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| nesoni                                 | latest                      | SHOULD             | http://bioinformatics.net.au/software.nesoni.shtml         |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| RAxML                                  | latest                      | SHOULD             | https://github.com/stamatak/standard-RAxML                 |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| phyml_v3                               | latest                      | SHOULD             |                                                            |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| paul                                   | latest                      | SHOULD             |                                                            |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| galaxy                                 | >ca0c4ad2bb39               | SHOULD             | http://galaxy.psu.edu/                                     |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| swarm                                  | latest                      | SHOULD             | http://biowulf.nih.gov/apps/swarm.html /download/swarm.tar |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| EMBOSS                                 | latest                      | SHOULD             | http://emboss.sourceforge.net/                             |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| WU-BLAST                               | latest                      | SHOULD             | http://blast.wustl.edu/                                    |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+
| BLAT                                   | latest                      | SHOULD             | http://genome.ucsc.edu/FAQ/FAQblat.html                    |
+----------------------------------------+-----------------------------+--------------------+------------------------------------------------------------+

This productivity environment will be supplemented with other related
productivity tools as they become available on allocated systems.

Potential sources of information for further future upgrades of this list include:
  * http://www.vital-it.ch/software/tools.php
  * http://www.csc.fi/english/research/sciences/bioscience/programs/index_html
  * http://bacpathgenomics.wordpress.com/software/
  * http://umbc.rnet.missouri.edu/general/software/alphabetical.html
  * http://www.broadinstitute.org/software/bsi-sig/
  * http://ncgas.org/software, genome assembly, assembly, genomics
  * http://confluence.rcs.griffith.edu.au:8080/display/GHPC/qiime#qiime-Alignment%2Ctreebuilding%2Ctaxonomyassignment%2COTUpicking%2Candotherdatagenerationsteps%28requiredforalternativepipelines%29
  * http://www.bioplexity.org/lectures/ebi-s10.pdf
  * http://helix.nih.gov/Applications/

Kindly notify -if this policy is inadequate for your work-
both your local site technical representative & HPCBIOS user-support.

